packages/url/src/index.js

```diff
@@ -0,0 +1,127 @@
+import mime from 'mime';
+import { createFilter } from 'rollup-pluginutils';
+
+const defaultInclude = ['**/*.svg', '**/*.png', '**/*.jpg', '**/*.gif'];
+
+export default function url(options = {}) {
+  const {
+    limit = 14 * 1024,
+    include = defaultInclude,
+    exclude,
+    publicPath = '',
+    emitFiles = true,
+    fileName = '[hash][extname]'
+  } = options;
+  const filter = createFilter(include, exclude);
+
+  const copies = Object.create(null);
+
+  return {
+    load(id) {
+      if (!filter(id)) {
+        return null;
+      }
+      return Promise.all([promise(fs.stat, id), promise(fs.readFile, id)]).then(
+        ([stats, buffer]) => {
+          let data;
+          if ((limit && stats.size > limit) || limit === 0) {
+            const hash = crypto
+              .createHash('sha1')
+              .update(buffer)
+              .digest('hex')
+              .substr(0, 16);
+            const ext = path.extname(id);
+            const name = path.basename(id, ext);
+            // Determine the directory name of the file based
+            // on either the relative path provided in options,
+            // or the parent directory
+            const relativeDir = options.sourceDir
+              ? path.relative(options.sourceDir, path.dirname(id))
+              : path
+                  .dirname(id)
+                  .split(path.sep)
+                  .pop();
+
+            // Generate the output file name based on some string
+            // replacement parameters
+            const outputFileName = fileName
+              .replace(/\[hash\]/g, hash)
+              .replace(/\[extname\]/g, ext)
+              .replace(/\[dirname\]/g, `${relativeDir}/`)
+              .replace(/\[name\]/g, name);
+            data = `${publicPath}${outputFileName}`;
+            copies[id] = outputFileName;
+          } else {
+            const mimetype = mime.getType(id);
+            const isSVG = mimetype === 'image/svg+xml';
+            data = isSVG ? encodeSVG(buffer) : buffer.toString('base64');
+            const encoding = isSVG ? '' : ';base64';
+            data = `data:${mimetype}${encoding},${data}`;
+          }
+          return `export default "${data}"`;
+        }
+      );
+    },
+    generateBundle: async function write(outputOptions) {
+      // Allow skipping saving files for server side builds.
+      if (!emitFiles) return;
+
+      const base = options.destDir || outputOptions.dir || path.dirname(outputOptions.file);
+
+      await makeDir(base);
+
+      Promise.all(
+        Object.keys(copies).map(async (name) => {
+          const output = copies[name];
+          // Create a nested directory if the fileName pattern contains
+          // a directory structure
+          const outputDirectory = path.join(base, path.dirname(output));
+          await makeDir(outputDirectory);
+          return copy(name, path.join(base, output));
+        })
+      );
+    }
+  };
+}
+
+function promise(fn, ...args) {
```

Comment:
that's rather surprising name - I would go with `promisify`, in fact there is `require('util').promisify` which already does this, but I'm not sure if it's available in all currently supported node versios, so that would have to be checked

Reply:
I had thought the same early on in this work but forgot to go back and check. Thanks for the reminder, util.promisify is from node 8 so it should be safe to use in this project, I've added it 👍

Reply:
@Andarist it's important to remember when reviewing migration PRs, that we're just trying to get the plugin moved in it's present form. We'll be opening issues to analyze each plugin after the entire migration is done.